hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -874,6 +875,8 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		} else {
 			log.Info("Reconciled MachineSet", "result", result)
 		}
+		// we use machineSet.Spec.Replicas because .Spec.Replicas will not be set if autoscaling is enabled
```

Comment:
I am afraid setting -1 would break any panels or alerts we setup based on this information. We could add a label to signal autoscaling being enabled.